pkgs/applications/networking/irc/tiny/default.nix

```diff
@@ -3,8 +3,9 @@
 , rustPlatform
 , fetchFromGitHub
 , pkg-config
-, dbus
-, openssl
+, dbusSupport ? stdenv.isLinux, dbus
+, rustlsPreferredToOpenssl ? !stdenv.isLinux, openssl
```

Comment:
```suggestion
, useOpenSSL ? stdenv.isLinux, openssl
```

Reply:
though is there a reason to prefer openssl on linux? when tiny was added there was no rustls support in tiny so thats why it was in buildInputs

Reply:
I made a poor choice of wording here.  Tiny doesn't have an option that says "you should use rustls".  It effectively assumes that the user prefers openssl, and uses openssl if it can find it.  I meant the option name to indicate that the user's preference is the opposite of what tiny assumes.

Now I can see that my name choice could be misinterpreted as blessing one library over the other, which is definitely not what I intended. 

I like your choice much better, I will include that in the commit.

The jury is still out regarding openssl vs rustls.  As is inevitable for cryptography libraries, rustls has quite a lot of assembler (wrapped in `unsafe` blocks) in it, so the memory-safety guarantees of "safe rust" are not as meaningful as they are in most situations.  Rustls is also far less battle-tested than openssl.  On the other hand, rustls is far simpler than openssl.